cpp/ql/src/experimental/Security/CWE/CWE-1126/DeclarationOfVariableWithUnnecessarilyWideScope.ql

```diff
@@ -0,0 +1,62 @@
+/**
+ * @name Errors When Using Variable Declaration Inside Loop
+ * @description Using variables with the same name is dangerous.
+ *              However, such a situation inside the while loop can lead to a violation of the accessibility of the program.
+ *              Requires the attention of developers.
+ * @kind problem
+ * @id cpp/errors-when-using-variable-declaration-inside-loop
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-1126
```

Comment:
I think this is more an issue of correctness than security, and I'm not convinced the CWE-1126 tag is accurate.

Reply:
The main influence vector I see is replacing the scope of a variable in the loop condition with the scope of a variable declared inside the loop.
this brings me to the real issue of security, resource depletion and `CWE-1126`.